Add Plugins to list of core source controlled paths to refresh

Damon3000s opened the pull request:
tldr; We've accidentally broken checking out files from the project plugin path within our fork, but the fix is compatible with the unmodified version and we believe still a correct change.

Context: We have a fork of this plugin and have made extensive changes in an effort to increase developer productivity without degrading the functionality of the plugin too much. As a result of this we accidentally broke the ability to checkout files from within the project plugins path. We resolved this and while this issue doesn't occur in the unmodified version of the plugin, we still believe that the changes made are correct and good.

The specifics as to what we changed and what broke is that the `git lfs locks` operation is really slow so we've tried to limit the number of times we call it for user initiated actions. The important location that was changed for this issue is that the Update Status action will no longer call `git lfs locks` and the LockState of the file will be `ELockState::Unset` and the context menu won't allow you to checkout the file (You can only check out lockable files that are `ELockState::NotLolcked` and up to date)

The fix is to make sure that the project plugins directory is checked in the various locations where we "update the SCC status of all packages", "update the status of all our files", etc. Previously it was only targeting the Content, Config and Project file paths so on startup and periodically it wouldn't get information about files in the plugins directory, but whenever we did Update Status on a single file (or folder) it would (in the unmodified version). By adding `FPaths::ProjectPluginsDir` to the various lists (and consolidating the lists) we ensure that the directory is checked when doing these project wide updates.

A bonus and related change is fixing some string literals pointing to the "Plugins/" and "Binaries/" directories. Our repo structure is such that we don't have those folders in the root, so those paths were never correct for us relative to the repo root. By using paths relative to the project dir we can ensure they are correct no matter how the repo is structured. 

We saw but didn't change anything related to the `.checksum` file because we don't have that file and aren't sure what it is or where it lives. We can deduce it exists in the project directory but didn't want to make changes that we can't verify.

Damon3000s committed:
Add Plugins to various lists of Source Control folders

Damon3000s committed:
Update another location which should check plugins folder

Damon3000s committed:
Fix for incorrect plugins path

Damon3000s committed:
Encapsulte the list of folders/files to refresh in git

Damon3000s committed:
FIx compile errors

Damon3000s committed:
Add descriptive comment for new function

Damon3000s committed:
Improve how we check for changes in the Plugins and Binaries path

- Git returns results that are repo relative and for our project that means "Project/Content/..." but the code was originally checking for `if Path.StartsWith("Plugins/")` which would never be true. Attempting to convert the Plugins path to be repo relative was not going well so it was easier to convert to absolute paths to compare, and we need the git results to be absolute paths later anyway
- .checksum is unchanged as we don't have that file in our project and don't know hown to handle it

Damon3000s committed:
Rename function

mastercoms commented:
Yes .checksum is relative to the project. It's generated by PBSync for Binaries syncing.

mastercoms reviewed (changes requested):
Looks great! Love the refactor and fixes this change does. Thanks for the contribution and detail in your PR! Just one change though, I'd love for the .checksum to be fixed now that this has been verified :)

Damon3000s committed:
Update the .checksum path to be relative to ProjectDir

- And ensure the comparison is correctly comparing the absolute paths

Damon3000s committed:
Rename AbsoluteProjectPath to make it clear this is the project DIRECTORY path, not the file (uproject) path.

Damon3000s committed:
More renaming to resolve any possible dir/file ambiguation

Damon3000s commented:
> Looks great! Love the refactor and fixes this change does. Thanks for the contribution and detail in your PR! Just one change though, I'd love for the .checksum to be fixed now that this has been verified :)

I've updated the code relating to the .checksum to be the same. I can't test it fully since we don't have the file, but it follows the same code as the other pieces I updated and tested earlier.

I also renamed some variables to disambiguate if the variable was a path to a directory or file.
